File: server/zlj2019/game_server/src/com/imop/lj/gameserver/quest/msg/CGFinishQuest.java
package com.imop.lj.gameserver.quest.msg;

import com.imop.lj.core.msg.MessageType;
import com.imop.lj.gameserver.common.msg.CGMessage;
import com.imop.lj.gameserver.quest.handler.QuestHandlerFactory;

/**
 * 完成任务
 * 
 * @author CodeGenerator, don't modify this file please.
 */
public class CGFinishQuest extends CGMessage{
	
	/** 任务Id */
	private int questId;
	
	public CGFinishQuest (){
	}
	
	public CGFinishQuest (
			int questId ){
			this.questId = questId;
	}
	
	@Override
	protected boolean readImpl() {

	// 任务Id
	int _questId = readInteger();
	//end



			this.questId = _questId;
		return true;
	}
	
	@Override
	protected boolean writeImpl() {

	// 任务Id
	writeInteger(questId);


		return true;
	}
	
	@Override
	public short getType() {
		return MessageType.CG_FINISH_QUEST;
	}
	
	@Override
	public String getTypeName() {
		return "CG_FINISH_QUEST";
	}

	public int getQuestId(){
		return questId;
	}
		
	public void setQuestId(int questId){
		this.questId = questId;
	}


	@Override
	public void execute() {
		QuestHandlerFactory.getHandler().handleFinishQuest(this.getSession().getPlayer(), this);
	}
}

File: u3u2/Assets/Scripts/app/net/c2s/CGFinishQuest.cs
using System;
using System.IO;
namespace app.net
{

/**
 * 完成任务
 * 
 * @author CodeGenerator, don't modify this file please.
 */
public class CGFinishQuest :BaseMessage
{
	
	/** 任务Id */
	private int questId;
	
	public CGFinishQuest ()
	{
	}
	
	public CGFinishQuest (
			int questId )
	{
			this.questId = questId;
	}
	
	protected override void ReadImpl() 
	{
		return;
	}
	
	protected override void WriteImpl() 
	{
	// 任务Id
	WriteInt(questId);

	}
	
	public override short GetMessageType()
	{
		return (short)MessageType.CG_FINISH_QUEST;
	}
	
	public override string getEventType()
	{
		return "";
	}
	
	}
}